### DISCUSSION TITLE
Now I use outlines+vllm and it takes 10 seconds to generate a JSON. This speed is not feasible in my current business. Is there any accelerated solution, such as a paper, plan or PR?

### DISCUSSION THREAD

**Discussion by lwdnxu** [outlines-dev/outlines#1161] [Q&A]
Now I use outlines+vllm and it takes 10 seconds to generate a JSON. This speed is not feasible in my current business. Is there any accelerated solution, such as a paper, plan or PR?

**Comment by just-cameron** [comment]
Speed improvements are always being worked on, but it may also be worth discussing your general setup. What models and schemas do you tend to work with? There might be some places we can suggest performance improvements.

**Comment by lwdnxu** [comment]
model： qwen2-7b
GPU:  4090 
I want to generate json like this :{
            "dataSource": "1002",
            "column": "200-279",
            "value": [
                "360"
            ],
            "mark": "16",
            "unit": null
        }
Number of tokens per request: 4166
Using vllm+ outlines, the speed is about 10+ seconds
Using vllm, the speed is about 1.x seconds

I would like to ask if there is an optimized solution? And what is the maximum number of seconds you can achieve in this case?

**Reply by lapp0** [reply]
@lwdnxu Each new schema requires an index to be compiled, which has some overhead. Once generation starts, overhead is ~1ms/token. This index only needs to be compiled once per unique schema.

Are you using a new schema each time, or are you seeing slowness re-using the same schema?
